Editor/EditorCore/PolyShapeTool.cs

```diff
@@ -1047,6 +1095,15 @@ void OnFinishVertexMovement()
 
         void UndoRedoPerformed()
         {
+            if (polygon != null && polygon.polyEditMode == PolyShape.PolyEditMode.Height)
+            {
+                //resetting state if undo is performed while in height edit mode
+                polygon.polyEditMode =  PolyShape.PolyEditMode.Path;
+                polygon.extrude = 0f;
+                polygon.mesh.Clear();
+                DestroyImmediate(polygon.mesh.mesh);
```

Comment:
```suggestion
                DestroyImmediate(polygon.mesh.mesh, true );
```

Fixes exception thrown after these steps:

1. Put some points
2. Undo a few
3. Redo all of the undone ones
4. Connect last one and start defining height
5. dont finalize height - undo. Throws...